exporter/awsprometheusremotewriteexporter/auth.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsprometheusremotewriteexporter
+
+import (
+	"bytes"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+// signingRoundTripper is a Custom RoundTripper that performs AWS Sig V4
+type signingRoundTripper struct {
```

Comment:
So keep in mind that I don't follow the details of OTel code or SIG discussions closely most of the time... I mostly interact with OTel at a higher design level... but it seems odd that this code is in a specific exporter. It feels like a generic AWS Auth module should be a core part of the OTel collector and that they should let all vendors create an auth module.

Also seems strange to me that we have our own version of this exporter, rather than the generic prom exporter having support to use any auth module (which would be a general feature of OTel exporters).

I'm just rambling... I am sure this was discussed in SIG meetings and argued over...

Reply:
Oh... I just went and read the discussion with Bogdan, and it seems like that's partly about moving this code into the core?

Reply:
From our discussions with upstream I believe that vendor specific code like this was simply not wanted in the core collector, hence why this logic is all going to the contrib repo. Since this logic must exist in the contrib repo there is no other option but to have another exporter since core components must not depend on contrib components.

A generic AWS Sigv4 module in the contrib repo that can be used by multiple exporters in the contrib repo however is something definitely worth considering